src/ngraph_sgcompiler.cc

```diff
@@ -37,6 +37,24 @@
 
 namespace ngraph_bridge {
 
+template <typename T>
+std::vector<bool> is_bool(const T& nodes) {
+  std::vector<bool> bools;
+  for (const auto& node : nodes) {
+    bools.push_back(node->get_element_type() == ngraph::element::boolean);
+  }
+  return bools;
+}
+template <typename T>
+std::vector<bool> is_scalar(const T& nodes) {
+  std::vector<bool> bools;
+  for (const auto& node : nodes) {
+    bools.push_back(node->get_shape().size() == 0);
```

Comment:
Is this one of those cases where the shape `[1]` should also be considered scalar?

Reply:
Yes, MXNet treats `[1]` as a scalar, and nGraph treats both `[]` and `[1]` as scalars. When doing the cache, we don't control which one nGraph uses, so this is to swap back and forth.